pkg/kafka/auth/tls.go

```diff
@@ -26,6 +26,8 @@ import (
 func setTLSConfiguration(config *tlscfg.Options, saramaConfig *sarama.Config, logger *zap.Logger) error {
 	if config.Enabled {
 		tlsConfig, err := config.Config(logger)
+		defer config.Close()
```

Comment:
what does `defer` do?

Reply:
it close `certWatcher` which uses `fswatcher` which was giving goleak issue that I talked about

Reply:
correct, but _when_ does it do that?

Reply:
after the test ends when tlscfg.Options is ended

Reply:
no. Read up on how defer works.